test/unit/AppLangSelectControllerSpec.js

```diff
@@ -0,0 +1,52 @@
+describe('AppLangSelectController', function () {
+  var $controller, $scope;
+
+  beforeEach(module('ui.bootstrap'));
+  beforeEach(module('izhukov.utils'));
+  beforeEach(module('myApp.i18n'));
+  beforeEach(module('myApp.services'));
+  beforeEach(module('myApp.controllers'));
+
+  beforeEach(function () {
```

Comment:
Again, AngularJS docs discourage [implicit annotation](https://docs.angularjs.org/guide/di#implicit-annotation).

Yes, the controllers knows about its arguments, but it does not know, where to get them from. The dependencies would need to be loaded before. The [inline array notation](https://docs.angularjs.org/guide/di#inline-array-annotation) works similiar to how AMD is structured by declaring it first and register a callback. Those strings can be collected in an object where a module loader can figure out the dependency tree and load the modules accordingly.

The third option would be the [$inject-property-annotation](https://docs.angularjs.org/guide/di#-inject-property-annotation). This requires changes at the code as well.

Maybe I can find a library which helps unit testing …